presto-native-execution/presto_cpp/main/operators/TestingPersistentShuffle.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/operators/TestingPersistentShuffle.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox;
+
+namespace {
+inline std::string createShuffleFilename(
+    const std::string& rootPath,
+    int32_t partition,
+    int fileIndex) {
+  return fmt::format("{}/{}_{}.bin", rootPath, partition, fileIndex);
+}
+
+/// This file is used to indidate that the shuffle system is ready to
+/// be used for reading (acts as a sync point between readers if needed).
+/// Mostly used for test purposes.
+const static std::string kReadyForReadFilename = "readyForRead";
+}; // namespace
+
+namespace facebook::presto::operators {
+
+std::unique_ptr<velox::WriteFile> TestingPersistentShuffle::getNextOutputFile(
+    int32_t partition) {
+  auto fileCount = getPartitionFilesCount(partition);
+  std::string filename = createShuffleFilename(rootPath_, partition, fileCount);
+  return fileSystem_->openFileForWrite(filename);
+}
+
+int TestingPersistentShuffle::getPartitionFilesCount(int32_t partition) const {
+  auto fileCount = 0;
+  bool fileExist = false;
+  do {
+    std::string filename =
+        createShuffleFilename(rootPath_, partition, fileCount);
+    if (!fileSystem_->exists(filename)) {
```

Comment:
Is it possible for another process / thread to create a file right after this check returned false?